pkgs/test_core/lib/src/runner/engine.dart

```diff
@@ -293,6 +302,29 @@ class Engine {
     return success;
   }
 
+  Future<Null> _gatherCoverage(LiveSuiteController controller) async {
+    if (_coverage == null) return;
+
+    final RunnerSuite suite = controller.liveSuite.suite;
+
+    if (!suite.platform.runtime.isDartVM) return;
+
+    // for some reason the observatoryUrl URI doesn't have the query parameters populated
+    final String isolateId = Uri.splitQueryString(
+        suite.environment.observatoryUrl.toString().split('?')[1])['isolateId'];
```

Comment:
I'd prefer to let the `Uri` class handle the split on `?`. Does it work to use the `query` field on the existing URI?

```suggestion
        suite.environment.observatoryUrl.query)['isolateId'];
```

Reply:
I can double check. I'm not 100% sure why but I do remember the query and queryParameters members of the class being missing in the URI.

Reply:
Yeah, the query is unpopulated in this case. I'll take some time this afternoon and try and figure out why, I 100% recognize that this is a little hacky as-is.

Reply:
Ah, it's because the URL is fragmented:
`http://127.0.0.1:54637/wJV71-d7700=/#/inspect?isolateId=isolates%2F980739093&objectId=libraries%2F%4022019165`

The URI parser expects the query to come before that fragment identifier (`/#/`).

@natebosch do you have a preferred way for dealing with this? I could just do a more null-safe version of this split, or we could add something to the `Environment` class, but that might be overkill.

Reply:
Oh weird. It's unfortunate that the observatory does stuff so strangely.

How about something like:

```dart
    final String isolateId =
        Uri.parse(suite.environment.observatoryUrl.fragment)
            .queryParameters['isolateId'];
```

This way we're making it explicit that we're dealing with the fragment part and treating it as if it's a URI itself. We won't end up with any trouble if there is a separate `?` in the URI.

I'm not sure if this makes us more brittle if the observatory changes it's URI patterns though.

Reply:
Any big changes like that would probably go over a major version update anyways, right? Like you said, it still has the potential to be fragile but I definitely agree that this is less fragile than the alternative. I also think this suggestion is much more clear about the intentions, like you said. I'll update it!

Reply:
Updated!